llmfoundry/utils/prompt_files.py

```diff
@@ -56,3 +68,73 @@ def load_prompts_from_file(prompt_path: str,
     if prompt_delimiter is None:
         return [prompt_string]
     return [i for i in prompt_string.split(prompt_delimiter) if i]
+
+
+def load_prompts_from_remote(prompt_path: str,
+                             prompt_delimiter: Optional[str] = None
+                            ) -> List[str]:
+    """Load a set of prompts from object storage.
+
+    Args:
+        prompt_path (str): Path for text file
+        prompt_delimiter (Optional str): Delimiter for text file
+            If not provided, assumes the prompt file is a single prompt (non-delimited)
+
+    Returns:
+        List of prompt string(s)
+    """
+    backend, _, _ = parse_uri(prompt_path)
+    if backend in ['', None]:
+        raise ValueError(
+            f'prompt_path_str must start with s3:// etc if using object storage'
+        )
+
+    local_path = prompt_path.split('/')[-1]
+    get_file(path=prompt_path, destination=local_path, overwrite=True)
+
+    with open(local_path, 'r') as f:
+        prompt_string = f.read()
+
+    if prompt_delimiter is None:
+        return [prompt_string]
+    return [i for i in prompt_string.split(prompt_delimiter) if i]
+
+
+def load_prompts_from_dataset(dataset_path: str,
+                              prompt_delimiter: Optional[str] = None
```

Comment:
I'd prefer you just add a new argument to the script for the dataset column name instead of overloading the delimiter param